contrib/format-spss/src/main/java/org/apache/drill/exec/store/spss/SpssBatchReader.java

```diff
@@ -0,0 +1,240 @@
+    try {
+      fsStream = negotiator.fileSystem().openPossiblyCompressedStream(split.getPath());
+      spssReader = new SpssDataFileReader(fsStream);
+    } catch (IOException e) {
+      throw UserException
+        .dataReadError(e)
+        .message("Unable to open SPSS File %s", split.getPath())
+        .addContext(e.getMessage())
+        .addContext(errorContext)
+        .build(logger);
+    }
+  }
+
+  private boolean processNextRow() {
+    try {
+      // Stop reading when you run out of data
+      if (!spssReader.readNextCase()) {
+        return false;
+      }
+
+      rowWriter.start();
+      for (SpssColumnWriter spssColumnWriter : writerList) {
+        spssColumnWriter.load(spssReader);
+      }
+      rowWriter.save();
+
+    } catch (IOException e) {
+      throw UserException
+        .dataReadError(e)
+        .message("Error reading SPSS File.")
+        .addContext(e.getMessage())
+        .addContext(errorContext)
+        .build(logger);
+    }
+    return true;
+  }
+
+  private TupleMetadata buildSchema() {
+    SchemaBuilder builder = new SchemaBuilder();
+    variableList = spssReader.getVariables();
+
+    for (SpssVariable variable : variableList) {
+      String varName = variable.getVariableName();
+
+      if (variable.isNumeric()) {
+        builder.addNullable(varName, TypeProtos.MinorType.FLOAT8);
+
+        // Check if the column has lookups associated with it
+        if (variable.getValueLabels() != null && variable.getValueLabels().size() > 0) {
+          builder.addNullable(varName + VALUE_LABEL, TypeProtos.MinorType.VARCHAR);
+        }
+
+      } else {
+        builder.addNullable(varName, TypeProtos.MinorType.VARCHAR);
+      }
+    }
+    return builder.buildSchema();
+  }
+
+  private void buildReaderList() {
+    writerList = new ArrayList<>();
+
+    for(SpssVariable variable : variableList) {
+      if (variable.isNumeric()) {
+        writerList.add(new NumericSpssColumnWriter(variable.getVariableName(), rowWriter, spssReader));
+      } else {
+        writerList.add(new StringSpssColumnWriter(variable.getVariableName(), rowWriter));
+      }
+    }
+  }
+
+  public abstract static class SpssColumnWriter {
+    final String columnName;
+
+    final ScalarWriter writer;
+
+    public SpssColumnWriter(String columnName, ScalarWriter writer) {
+      this.columnName = columnName;
+      this.writer = writer;
+    }
+
+
+    public abstract void load (SpssDataFileReader reader);
+  }
+
+  public static class StringSpssColumnWriter extends SpssColumnWriter {
+
+    StringSpssColumnWriter (String columnName, RowSetLoader rowWriter) {
+      super(columnName, rowWriter.scalar(columnName));
+    }
+
+    @Override
+    public void load(SpssDataFileReader reader) {
+      writer.setString(reader.getStringValue(columnName));
+    }
+  }
+
+  public static class NumericSpssColumnWriter extends SpssColumnWriter {
+
+    ScalarWriter labelWriter;
+
+    Map<Double, String> labels;
```

Comment:
This makes me nervous: `double` is a fragile thing to map from. Does SPSS require that indexed columns have integer values? If so, map from an `Integer`, which is more reliable.

Reply:
This is weird, but unfortunately, the only data types this format supports are strings and numerics which are floating point numbers.